public/js/widget_config.js

```diff
@@ -0,0 +1,382 @@
+  // TODO: Instead of using upper case widget names for unique-ness, use the
+  // TODO: rq widget namespace.
+  const widgetTypeUpper = objWidget.type.toUpperCase()
+
+  // TODO: Figure out how to replace the string 'widget' with a constant
+  const widgetContainer = jQuery(
+    `<div class="widget ${widgetTypeUpper}" id="${objWidget.label}"></div>`
+  )
+  const widgetHeader = jQuery(
+    '<div class="widget-header">' + objWidget.label + '</div>'
+  )
+  const widgetKebobMenu = jQuery(
+    '<img class="widget-kebobMenu" src="img/kebobMenu.png"/>'
+  )
+  const widgetContent = jQuery(
+    '<div class="widget-content"></div>'
+  )
+
+  widgetKebobMenu.appendTo(widgetContainer)
+  widgetHeader.appendTo(widgetContainer)
+  widgetContent.appendTo(widgetContainer)
+
+  if (Object.hasOwn(objWidget, 'keys')) {
+    keyControl.addKeysForWidget(objWidget) // eslint-disable-line no-undef
+  }
+
+  /*
+   * Store the widget configuration object (objWidget) under the WIDGET_NAMESPACE
+   * in an arbetrary jQuery data unit attatched to the widgetContainer
+   */
+  jQuery(widgetContainer).data(RQ_PARAMS.WIDGET_NAMESPACE, objWidget)
+
+  console.debug(`${JSON.stringify(objWidget)}`)
+
+  jQuery(widgetContainer)[widgetTypeUpper](
+    { ...objWidget, socket: socket }
+  ).appendTo(
+    '#widgets'
+  ).draggable({
+    handle: '.widget-header',
+    snap: true,
+    start: function (event, ui) {
+      const widgetId = event.currentTarget.id
+      console.debug(
+        `drag started on ${widgetId}` +
+        ` at position ${JSON.stringify(jQuery('#' + widgetId).position())}` +
+        ` and offset ${JSON.stringify(jQuery('#' + widgetId).offset())}`
+      )
+    },
+    stop: function (event, ui) {
+      const widgetId = event.target.id
+      const widgetData = jQuery('#' + widgetId).getWidgetConfiguration()
+
+      if (!widgetData) {
+        /*
+         * The widgetData no longer exists, likely because the widget was deleted from the UI. That condition
+         * is acceptable here.
+         */
+        return
+      }
+
+      const widgetPosition = widgetData.position
+
+      console.debug(
+        `drag stopped on ${widgetId}` +
+        ` at offset ${JSON.stringify(jQuery('#' + widgetId).offset())}` +
+        ` and position ${JSON.stringify(jQuery('#' + widgetId).position())}` +
+        ` original ${JSON.stringify(widgetPosition)}`
+      )
+
+      jQuery('#' + widgetId).getWidgetConfiguration().position = updateWidgetPosition(
+        widgetPosition,
+        jQuery('#' + widgetId).position()
+      )
+      console.debug(
+        ` updated position to: ${JSON.stringify(jQuery('#' + widgetId).getWidgetConfiguration().position)}`
+      )
+    }
+  }).hover(function (event) {
+    jQuery(event.currentTarget).find('.widget-kebobMenu')[0].style.display = 'block'
+  },
+  function (event) {
+    jQuery(event.currentTarget).find('.widget-kebobMenu')[0].style.display = 'none'
+  })
+
+  /*
+  * Two alternative ways to do the same thing.
```

Comment:
Couple of comments here.
1. I believe the common spelling is "kebab". See [Difference between hamburger and kebab](https://ux.stackexchange.com/questions/115468/what-is-the-difference-between-these-2-menu-icons-3-dots-kebab-and-3-lines-h)
2. More importantly, I don't see any kind of "menu" when I click the "kebab" - I'm simply taken to the 'Configure a widget' dialog.
3. Would an always-visible 'pencil' be simpler and more intuitive than any of:
    1. kebab
    2. hamburger
    3. falafel
    4. double-click